packages/nest-commander/README.md

```diff
@@ -48,6 +48,93 @@ Often times you're not just running a single command with a single input, but ra
+
+| property | type | required | description |
+| --- | --- | --- | --- |
+| name | string | true | The name that will be used to determine which `@Question()` this decorator belongs to. |
+
+##### Passing to the `@Question()` decorator
+
+Below is an example of using the `validate` method in the `@Question()` decorator
+
+```ts
+@Question({
+  type: 'input',
+  name: 'phone',
+  message: "What's your phone number?",
+  validate: function(value: string) {
+    const pass = value.match(
+      /^([01]{1})?[-.\s]?\(?(\d{3})\)?[-.\s]?(\d{3})[-.\s]?(\d{4})\s?((?:#|ext\.?\s?|x\.?\s?){1}(?:\d+)?)?$/i,
+    );
+    if (pass) {
+      return true;
+    }
+    return 'Please enter a valid phone number';
+  }
+})
+parsePhone(val: string) {
+  return val;
+}
+```
+
+##### Using the `@*For()` decorator
+
+Below is an example of a `@Question()` and `@ValidateFor()` decorator in use
+
+```ts
+@Question({
+  type: 'input',
+  name: 'phone',
+  message: "What's your phone number?",
+})
+parsePhone(val: string) {
+  return val;
+}
+
+@ValidateFor({ name: 'phone' })
+validatePhone(value: string) {
+  const pass = value.match(
+    /^([01]{1})?[-.\s]?\(?(\d{3})\)?[-.\s]?(\d{3})[-.\s]?(\d{4})\s?((?:#|ext\.?\s?|x\.?\s?){1}(?:\d+)?)?$/i,
+  );
+  if (pass) {
+    return true;
+  }
+
+  return 'Please enter a valid phone number';
+}
+```
+
+As you can see, the `name` of both `@Question()` and `@ValidateFor()` align, allowing the underlying engine to properly map the `validatePhone` method to the `phone`'s property set.
+
+^ Please note that `@*For()` is shorthand for `@ValidateFor()`, `@ChoicesFor()`, `@MessageFor()`, `@DefaultFor()`, and `@WhenFor()`.
+
+### InquirerService
+
+The `InquirerService` is an injectable provider that allows you to call inquirer for a specific set of questions (named with `@QuestionSet()`). When calling the question set, you can pass in the already obtained options as well, and inquirer will skip over the options that are already answered, unless the `askAnswered` property is set to `true` ass mentioned in their docs. You can use either `InquirerService#ask` or `InquirerService#prompt`, as they are aliases for each other. The return from the `InquirerService#prompt` method is the non-partial variant of the options passed in; in other words, the return is the answers that the user provided, mapping appropriately in the cases where necessary, such as lists. For an example usage, please check the [pizza integration test](../../integration/pizza/src/pizza.command.ts).
+
 ## Running the Command
 
 Similar to how in a NestJS application we can use the `NestFactory` to create a server for us, and run it using `listen`, the `nest-commander` package exposes a simple to use API to run your server. Import the `CommandFactory` and use the `static` method `run` and pass in the root module of your application. This would probably look like below
```

Comment:
Replace sentence: 

Similar to how, in a NestJS application, we can use the `NestFactory` to create a server for us and run it using `listen`, the `nest-commander` package also exposes a simple to use API to run your server.

Scott